client/extensions/relatedArticles/src/EventsRelatedArticles/EventsRelatedArticlesModal.tsx

```diff
@@ -0,0 +1,307 @@
+                contentBg="medium"
+                size="x-large"
+                onHide={closeModal}
+                footerTemplate={(
+                    <Spacer h gap="8" noWrap alignItems="end" justifyContent="end">
+                        <Button
+                            onClick={() => {
+                                closeModal();
+                            }}
+                            text={gettext('Cancel')}
+                            style="filled"
+                        />
+                        <Button
+                            onClick={() => {
+                                this.props.onChange(cleanArticlesFields(this.state.currentlySelectedArticles ?? []))
+                                closeModal();
+                            }}
+                            disabled={JSON.stringify(this.props.selectedArticles) === JSON.stringify(this.state.currentlySelectedArticles)}
+                            text={gettext('Apply')}
+                            style="filled"
+                            type="primary"
+                        />
+                    </Spacer>
+                )}
+            >
+                <Spacer v gap="0">
+                    <div
+                        style={{
+                            padding: 12,
+                            backgroundColor: 'white'
+                        }}
+                    >
+                        <SearchBar
+                            value={this.state.searchQuery}
+                            onSubmit={(value: string) => {
+                                this.setState({
+                                    searchQuery: value,
+                                });
+                            }}
+                            placeholder={gettext('Search...')}
+                            boxed
+                        >
+                            <Dropdown
+                                maxHeight={300}
+                                append
+                                zIndex={2001}
+                                items={[
+                                    {
+                                        type: 'group',
+                                        items: allLanguages.map((language) => ({
+                                            label: language.label,
+                                            onSelect: () => onSelectLanguage(language)
+                                        }))
+                                    },
+                                ]}
+                            >
+                                {this.state.activeLanguage.label}
+                            </Dropdown>
+                        </SearchBar>
+                    </div>
+                    <ContentDivider margin='none' />
+                    <Spacer h gap="0" justifyContent='start' alignItems='start' noWrap>
+                        <div style={{...{width: this.state.previewItem ? '65%' : '100%'}, ...{padding: 12}}}>
```

Comment:
Surely the width here can be done better, rather than constant percentage values (@fritzSF)?
Again with the padding, please use the helper classes